mobile/lib/shared/services/immich_logger.service.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:io';
+
+import 'package:flutter/widgets.dart';
+import 'package:hive/hive.dart';
+import 'package:immich_mobile/constants/hive_box.dart';
+import 'package:immich_mobile/shared/models/in_app_logger_message.model.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:share_plus/share_plus.dart';
+
+class ImmichLogger {
```

Comment:
Maybe we want to use https://pub.dev/packages/logging instead and configure it to write all log messages to the Hive box?

Reply:
Yes, I was looking at variety of logging library and they have all have similar and probably miss this one. I found of one the most widely used logging library that is recently getting abandoned, so I approach the PR with no logging dependency at all. However the library you showed looks very good and simple enough to drop in and out of our service